yellowbrick/features/jointplot.py

```diff
@@ -14,308 +14,369 @@
 ## Imports
 ##########################################################################
 
-import warnings
 import numpy as np
-import matplotlib as mpl
 import matplotlib.pyplot as plt
 
-from yellowbrick.features.base import FeatureVisualizer
-from yellowbrick.exceptions import YellowbrickValueError
-from yellowbrick.bestfit import draw_best_fit
-from yellowbrick.utils import is_dataframe
+try:
+    # Only available in Matplotlib >= 2.0.2
+    from mpl_toolkits.axes_grid1 import make_axes_locatable
+except ImportError:
+    make_axes_locatable = None
 
+from scipy.stats import pearsonr
+from .base import FeatureVisualizer
+# from ..bestfit import draw_best_fit
+from ..exceptions import YellowbrickValueError
 
+
+# Default Colors
 FACECOLOR = "#FAFAFA"
 HISTCOLOR = "#6897bb"
 
 
+# Objects for export
+__all__ = [
+    "JointPlot", "JointPlotVisualizer", "joint_plot",
+]
+
+
 ##########################################################################
 ## Joint Plot Visualizer
 ##########################################################################
 
 
-class JointPlotVisualizer(FeatureVisualizer):
+class JointPlot(FeatureVisualizer):
     """
```

Comment:
Still need to finish the docstring.

Reply:
Some text you can use as a start if you'd like:

Joint plots are useful for machine learning on multi-dimensional data, allowing for the visualization of complex interactions between different data dimensions, their varying distributions, and even their relationships to the target variable for prediction.  The Yellowbrick `JointPlot` can be used for both pairwise feature analysis and for feature-to-target plots. For pairwise feature analysis, the `columns` argument can be used to specify the indexes of the two desired columns in `X`. If `y` is also specified, the plot can be colored with a heatmap or by class. For feature-to-target plots, the user can provide either `X` and `y` as 1D vectors, or a `columns` argument with an index to a single feature in `X` to be plotted against `y`. Histograms can be included by setting the `hist` argument to `True` for a frequency distribution, or to `density` for a probability density function.